### PR TITLE
Finish user module

### PR DESCRIPTION
**Pull request by Nexucis**
* add permission
* inject permission in security management
* protect user account and allow only the owner account to get it or to delete it.
* provide update user endpoint
* add new Generic Mapper
* provide new endpoint to give available permissions
* add swagger

⚠️ this merge request must be accepted after the #10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # coddy
 [![Build Status](https://travis-ci.org/Crunchy-Torch/coddy.svg?branch=master)](https://travis-ci.org/Crunchy-Torch/coddy)
 
+## API
+
+If you want to see all endpoint provide by our api, you can take a loot at [this swagger](./docs/api-swagger.yml), or you can access through the [swaggerHub](https://app.swaggerhub.com/apis/Nexucis/Coddy/1.0.0)
```
Comment (DavidLevayer) [inline README.md]: Take a loot?

File: back/src/main/java/org/crunchytorch/coddy/application/data/ApiName.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.crunchytorch.coddy.application.data;
+
+public class ApiName {
```
Comment (DavidLevayer) [inline back/src/main/java/org/crunchytorch/coddy/application/data/ApiName.java]: `ApiConst `ou `ApiNameConst `?

Comment (DavidLevayer) [reply back/src/main/java/org/crunchytorch/coddy/application/data/ApiName.java]: En fait je retirerai carrément cette classe. Le `USER_LOGIN_PATH_PARAM` tu le mets dans User.java (en constante). Les autres, tu les laisses en dur dans les annotations.

File: back/src/main/java/org/crunchytorch/coddy/user/data/UpdateUser.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package org.crunchytorch.coddy.user.data;
+
+public class UpdateUser {
+
+    private String login;
+    /**
+     * a String type is not security safe. If you dumb the java memory, you will see clearly the password
```
Comment (DavidLevayer) [inline back/src/main/java/org/crunchytorch/coddy/user/data/UpdateUser.java]: dumb... dump?

> String type is not secure enough ; clear password can be seen through a JVM memory dump

**Review by DavidLevayer — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Nexucis) [reply README.md]: voilà c'est fait

File: back/src/main/java/org/crunchytorch/coddy/application/data/ApiName.java (hunk 2 above)
Comment (Nexucis) [reply back/src/main/java/org/crunchytorch/coddy/application/data/ApiName.java]: la classe n'existe plus ouais

**Review by DavidLevayer — APPROVED** [review]

**Pull request merged**